src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractAddItemCommandHandler.cs

```diff
@@ -0,0 +1,131 @@
+        private readonly IVsUIService<IVsAddProjectItemDlg> _addItemDialog;
+        private readonly IVsService<SVsShell, IVsShell> _vsShell;
+        private readonly Lazy<Dictionary<long, CommandDetails>> _cSharpCommandMap;
+        private readonly Lazy<Dictionary<long, CommandDetails>> _vbCommandMap;
+
+        [ImportingConstructor]
+        public AbstractAddItemCommandHandler(IPhysicalProjectTree projectTree, IUnconfiguredProjectVsServices projectVsServices, IVsUIService<IVsAddProjectItemDlg> addItemDialog, IVsService<SVsShell, IVsShell> vsShell)
+        {
+            _projectTree = projectTree;
+            _projectVsServices = projectVsServices;
+            _addItemDialog = addItemDialog;
+            _vsShell = vsShell;
+
+            _cSharpCommandMap = new Lazy<Dictionary<long, CommandDetails>>(() => GetCSharpCommands());
+            _vbCommandMap = new Lazy<Dictionary<long, CommandDetails>>(() => GetVBCommands());
+        }
+
+        protected abstract Dictionary<long, CommandDetails> GetCSharpCommands();
+        protected abstract Dictionary<long, CommandDetails> GetVBCommands();
+
+        public Task<CommandStatusResult> GetCommandStatusAsync(IImmutableSet<IProjectTree> nodes, long commandId, bool focused, string commandText, CommandStatus progressiveStatus)
+        {
+            Requires.NotNull(nodes, nameof(nodes));
+
+            if (TryGetCommandDetails(commandId, out _))
+            {
+                return GetCommandStatusResult.Handled(commandText, CommandStatus.Enabled);
+            }
+
+            return GetCommandStatusResult.Unhandled;
+        }
+
+        public async Task<bool> TryHandleCommandAsync(IImmutableSet<IProjectTree> nodes, long commandId, bool focused, long commandExecuteOptions, IntPtr variantArgIn, IntPtr variantArgOut)
+        {
+            Requires.NotNull(nodes, nameof(nodes));
+
+            if (nodes.Count == 1)
+            {
+                if (TryGetCommandDetails(commandId, out CommandDetails result))
+                {
+                    IProjectTree node = nodes.First();
+
+                    // Get the strings from the legacy package
+                    IVsShell vsShell = await _vsShell.GetValueAsync();
+                    ErrorHandler.ThrowOnFailure(vsShell.LoadPackageString(ref result.DirNamePackageGuid, result.DirNameResourceId, out string dirName));
+                    ErrorHandler.ThrowOnFailure(vsShell.LoadPackageString(ref result.TemplateNamePackageGuid, result.TemplateNameResourceId, out string templateName));
+
+                    return await ShowAddProjectItemDialog(node, dirName, templateName, _projectTree, _projectVsServices, _addItemDialog);
+                }
+            }
+
+            return false;
+        }
+
+        private bool TryGetCommandDetails(long commandId, out CommandDetails result)
+        {
+            IProjectCapabilitiesScope capabilities = _projectVsServices.ActiveConfiguredProject.Capabilities;
```

Comment:
This service lives at the "configuredproject" scope (look at the [ProjectSystemContract] applied to it), it should be importing ConfiguredProject and use that to filter, not the active configuration. Only unconfigured-project scoped things should use the active config.

Reply:
I certainly didn't check this, or think to, so my bad, but I think I might have fallen arse backwards into being correct anyway?

https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/IUnconfiguredProjectVsServices.cs#L11

Reply:
@davkean Can you confirm if this needs to be changed or not?

Reply:
Yes it still is a problem, not sure what you are pointing me to?

Reply:
Ah, I think I understand the confusion - I think you want to read up on scopes, and this in particular: https://github.com/Microsoft/VSProjectSystem/blob/master/doc/overview/scopes.md#scopes-at-odds.

Reply:
My bad, thought by "this service" you meant `_projectVsServices`. Will update as appropriate.

Reply:
Fixed